modules/series-service-impl/src/main/java/org/opencastproject/series/impl/persistence/SeriesServiceDatabaseImpl.java

```diff
@@ -383,6 +402,63 @@ public DublinCoreCatalog getSeries(String seriesId) throws NotFoundException, Se
     }
   }
 
+  @Override
+  public List<Series> getAllForAdministrativeRead(Date from, Optional<Date> to, int limit)
+          throws SeriesServiceDatabaseException {
+    // Validate parameters
+    if (limit <= 0) {
+      throw new IllegalArgumentException("limit has to be > 0");
+    }
+
+    // Make sure the user is actually an administrator of sorts
+    User user = securityService.getUser();
+    if (!user.hasRole(GLOBAL_ADMIN_ROLE) && !user.hasRole(user.getOrganization().getAdminRole())) {
+      throw new SeriesServiceDatabaseException(
+          new UnauthorizedException(user, getClass().getName() + ".getModifiedInRangeForAdministrativeRead")
+      );
+    }
+
+    // Load series from DB.
+    EntityManager em = emf.createEntityManager();
+    try {
+      TypedQuery<SeriesEntity> q;
+      if (to.isPresent()) {
+        if (from.after(to.get())) {
+          throw new IllegalArgumentException("`from` is after `to`");
+        }
```

Comment:
The Tobira code does not benefit from this function returning an empty list. Thus I opted for throwing an error for now as this might well be a bug in the caller code.